src/sentry/snuba/sessions_v2.py

```diff
@@ -421,6 +425,14 @@ def _run_sessions_query(query):
     `totals` and again for the actual time-series data grouped by the requested
     interval.
     """
+
+    # We only return the top-N groups, based on the first field that is being
+    # queried, assuming that those are the most relevant to the user.
+    # In a future iteration we might expose an `orderBy` query parameter.
+    orderby = [f"-{query.primary_column}"]
+
+    max_groups = SNUBA_LIMIT // len(get_timestamps(query))
```

Comment:
We talked about a fixed limit of 50, but this would not allow us to query more than 200 time intervals (50 * 200 = 10000), so I'd rather make it dynamic.

Reply:
I agree. I think this approach is better
I thought however you are developing this (initially atleast) for the specific use case in the UI where you request 50 releases and 90 data pts -> 50*90 = 4500, and we bump snuba limit to 10,000